### PR TITLE
Amends page.open to provide useful feedback on error

### PR DESCRIPTION
**Pull request by nickcall**
If the page.open fails we now provide a reason

### CODE REVIEW

**Comment by hollsk** [comment]
✨ SUPER DUPER ✨ 

This will help with debugging at least 2 or 3 issues in Pa11y, so keen to get it out 👍

File: lib/truffler.js [hunk 1]
```diff
@@ -141,16 +141,21 @@ Truffler.prototype._run = function(url, options, done) {
 		// Open the URL in PhantomJS
 		openUrl: guardAgainstTimeout(function(next) {
 			options.log.debug('Opening "' + url + '" in PhantomJS');
+
+			state.page.onResourceError = function(resourceError) {
+				this.reason = resourceError.errorString;
```
Comment (rowanmanning) [inline lib/truffler.js]: I'm not sure how `this` here is set to the right value, looks a bit magic. Could we maybe just initialise a `var` in the function above it and set that instead? Then we wouldn't need the `bind` on line 158.

Comment (nickcall) [inline lib/truffler.js]: I worked with @moddular on that bit of code, we tried using a var in the beginning and it wouldn't work so ended up with the rather odd bind and this.

Comment (moddular) [inline lib/truffler.js]: Yeah, it is a bit ugly. The problem we had with using `var` was that the scoping seemed to be wrong when the code ran in the tests, making `reason` always `undefined` there. I wonder if there's some tweaking of the test setup we could do to resolve this without needing `this`/`bind`?

Comment (rowanmanning) [inline lib/truffler.js]: In that case let's leave it as is for now and we can maybe refactor later 👍

**Comment by hollsk** [comment]
@nickcall can you merge and tag/release this, please? We'll be able to update/close off a few Pa11y issues once you do 👍

**Pull request merged**